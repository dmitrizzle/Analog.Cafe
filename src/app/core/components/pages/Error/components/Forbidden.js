import React from "react"

import { withRouter } from "react-router"

import { HEADER_ERRORS, TEXT_ERRORS } from "../../../../../constants"
import ErrorPage from "../../../vignettes/ErrorPage"
import Link from "../../../controls/Link"

class Forbdden extends React.PureComponent {
  componentWillMount = () => {
    this.props.history.replace({
      state: {
        status: "403"
      }
    })
  }
  componentWillUnmount = () => {
    this.props.history.replace({
      state: {
        status: "200"
      }
    })
  }
  render = () => {
    return (
      <ErrorPage
        errorTitle={HEADER_ERRORS.ARTICLE.title}
        errorSubtitle={HEADER_ERRORS.ARTICLE.subtitle}
        errorDetails={TEXT_ERRORS.CODE_403.error}
      >
        You need to{" "}
        <strong>
          <Link to="/sign-in">sign in</Link>
        </strong>{" "}
        to view this page.
      </ErrorPage>
    )
  }
}

export default withRouter(Forbdden)
