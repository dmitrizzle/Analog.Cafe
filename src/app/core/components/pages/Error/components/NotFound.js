import React from "react"

import { HEADER_ERRORS, TEXT_ERRORS } from "../../../../../constants"
import Error from ".."
import Link from "../../../controls/Link"

export default class extends React.PureComponent {
  componentWillMount = () => {
    try {
      this.props.history.replace({
        state: {
          status: "404"
        }
      })
    } catch (exception) {
      console.log(
        "Couldn't push to history. You probably used two slashes instead of one in URL."
      )
    }
  }
  componentWillUnmount = () => {
    this.props.history.replace({
      state: {
        status: "200"
      }
    })
  }
  render = () => {
    return (
      <Error
        errorTitle={HEADER_ERRORS.ARTICLE.title}
        errorSubtitle={HEADER_ERRORS.ARTICLE.subtitle}
        errorDetails={TEXT_ERRORS.CODE_404.error}
      >
        Click{" "}
        <strong>
          <Link to="/">here</Link>
        </strong>{" "}
        to go to homepage.
      </Error>
    )
  }
}
