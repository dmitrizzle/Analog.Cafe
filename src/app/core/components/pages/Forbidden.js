import { Helmet } from "react-helmet"
import React from "react"

import { withRouter } from "react-router"

import { HEADER_ERRORS, TEXT_ERRORS } from "../../../constants"
import ArticleSection from "./Article/components/ArticleSection"
import ArticleWrapper from "./Article/components/ArticleWrapper"
import HeaderLarge from "../vignettes/HeaderLarge"
import Link from "../controls/Link"

class NotFound extends React.PureComponent {
  componentWillMount = () => {
    this.props.history.replace({
      state: {
        status: "403"
      }
    })
  }
  componentWillUnmount = () => {
    this.props.history.replace({
      state: {
        status: "200"
      }
    })
  }
  render = () => {
    return (
      <ArticleWrapper>
        <Helmet>
          <title>{HEADER_ERRORS.ARTICLE.title}</title>
        </Helmet>
        <HeaderLarge
          pageTitle={HEADER_ERRORS.ARTICLE.title}
          pageSubtitle={HEADER_ERRORS.ARTICLE.subtitle}
          title={TEXT_ERRORS.CODE_403.error}
        />{" "}
        <ArticleSection>
          <p style={{ textAlign: "center" }}>
            You need to{" "}
            <strong>
              <Link to="/sign-in">sign in</Link>
            </strong>{" "}
            to view this page.
          </p>
        </ArticleSection>
      </ArticleWrapper>
    )
  }
}

export default withRouter(NotFound)
