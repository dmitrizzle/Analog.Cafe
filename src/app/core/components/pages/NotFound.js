import React from "react"

import { Article, Section } from "../styles/ArticleStyles"
import Heading from "../vignettes/ArticleHeading"
import Helmet from "../vignettes/Helmet"
import Link from "../controls/Link"
import errorMessages from "../../../user/constants/errors"

export default class extends React.PureComponent {
  componentWillMount = () => {
    this.props.history.replace({
      state: {
        status: "404"
      }
    })
  }
  componentWillUnmount = () => {
    this.props.history.replace({
      state: {
        status: "200"
      }
    })
  }
  render = () => {
    return (
      <Article>
        <Helmet>
          <title>{errorMessages.VIEW_TEMPLATE.ARTICLE.title}</title>
        </Helmet>
        <Heading
          pageTitle={errorMessages.VIEW_TEMPLATE.ARTICLE.title}
          pageSubtitle={errorMessages.VIEW_TEMPLATE.ARTICLE.subtitle}
          title={errorMessages.DISAMBIGUATION.CODE_403.error}
        />
        <Section>
          <p style={{ textAlign: "center" }}>
            Click{" "}
            <strong>
              <Link to="/">here</Link>
            </strong>{" "}
            to go to homepage.
          </p>
        </Section>
      </Article>
    )
  }
}
// NOTE: withRouter() props inherited from /components/_screens/AppRoutes
