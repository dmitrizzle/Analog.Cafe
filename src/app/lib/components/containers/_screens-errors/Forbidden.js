// tools
import React from "react"
import { withRouter } from "react-router"
import Helmet from "../../stateless/_async/Helmet"

// components
import Link from "../../stateless/_controls/Link"
import Heading from "../../stateless/ArticleHeading"
import { Article, Section } from "../../stateless/ArticleStyles"

import errorMessages from "../../../constants/messages/errors"

class NotFound extends React.PureComponent {
  componentWillMount = () => {
    this.props.history.replace({
      state: {
        status: "403"
      }
    })
  }
  componentWillUnmount = () => {
    this.props.history.replace({
      state: {
        status: "200"
      }
    })
  }
  render = () => {
    return (
      <Article>
        <Helmet>
          <title>{errorMessages.VIEW_TEMPLATE.ARTICLE.title}</title>
        </Helmet>
        <Heading
          pageTitle={errorMessages.VIEW_TEMPLATE.ARTICLE.title}
          pageSubtitle={errorMessages.VIEW_TEMPLATE.ARTICLE.subtitle}
          title={errorMessages.DISAMBIGUATION.CODE_403.error}
        />{" "}
        <Section>
          <p style={{ textAlign: "center" }}>
            You need to{" "}
            <strong>
              <Link to="/sign-in">sign in</Link>
            </strong>{" "}
            to view this page.
          </p>
        </Section>
      </Article>
    )
  }
}

export default withRouter(NotFound)
