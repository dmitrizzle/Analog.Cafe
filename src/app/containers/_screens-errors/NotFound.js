// tools
import React from "react"
import Link from "../../components/Link"
import Helmet from "react-helmet"

// components
import Heading from "../../components/ArticleHeading"
import { Section, Article } from "../../components/ArticleStyles"

import errorMessages from "../../../constants/messages/errors"

// render
export default class extends React.PureComponent {
  componentWillMount = () => {
    this.props.history.replace({
      state: {
        status: "404"
      }
    })
  }
  componentWillUnmount = () => {
    this.props.history.replace({
      state: {
        status: "200"
      }
    })
  }
  render = () => {
    return (
      <Article>
        <Helmet>
          <title>{errorMessages.VIEW_TEMPLATE.ARTICLE.title}</title>
        </Helmet>
        <Heading
          pageTitle={errorMessages.VIEW_TEMPLATE.ARTICLE.title}
          pageSubtitle={errorMessages.VIEW_TEMPLATE.ARTICLE.subtitle}
          title={errorMessages.DISAMBIGUATION.CODE_403.error}
        />
        <Section>
          <p style={{ textAlign: "center" }}>
            Click{" "}
            <strong>
              <Link to="/">here</Link>
            </strong>{" "}
            to go to homepage.
          </p>
        </Section>
      </Article>
    )
  }
}
// NOTE: withRouter() props inherited from /components/_screens/AppRoutes
