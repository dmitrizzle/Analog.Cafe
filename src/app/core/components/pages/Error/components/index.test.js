import { shallow } from "enzyme"
import React from "react"

import Forbidden from "./Forbidden"
import NotFound from "./NotFound"

it("Render Forbidden without crashing", () => {
  shallow(<Forbidden />)
})

it("Render NotFound without crashing", () => {
  shallow(
    <NotFound
      history={{
        replace: () => {}
      }}
    />
  )
})
