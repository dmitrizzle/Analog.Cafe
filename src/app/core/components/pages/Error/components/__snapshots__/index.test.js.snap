// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Forbidden, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Forbdden) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Render NotFound, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PureComponent
    history={
        Object {
            "replace": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        "Click",
        " ",
        <strong>
          <Unknown
                    to="/"
          >
                    here
          </Unknown>
</strong>,
        " ",
        "to go to homepage.",
      ],
      "errorDetails": "Error: This view or data for this view does not exist. (404)",
      "errorSubtitle": "Page Not Available",
      "errorTitle": "ʕ⊙ᴥ⊙ʔ",
    },
    "ref": null,
    "rendered": Array [
      "Click",
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            to="/"
>
            here
</Unknown>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "here",
            "to": "/",
          },
          "ref": null,
          "rendered": "here",
          "type": [Function],
        },
        "type": "strong",
      },
      " ",
      "to go to homepage.",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          "Click",
          " ",
          <strong>
            <Unknown
                        to="/"
            >
                        here
            </Unknown>
</strong>,
          " ",
          "to go to homepage.",
        ],
        "errorDetails": "Error: This view or data for this view does not exist. (404)",
        "errorSubtitle": "Page Not Available",
        "errorTitle": "ʕ⊙ᴥ⊙ʔ",
      },
      "ref": null,
      "rendered": Array [
        "Click",
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              to="/"
>
              here
</Unknown>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "here",
              "to": "/",
            },
            "ref": null,
            "rendered": "here",
            "type": [Function],
          },
          "type": "strong",
        },
        " ",
        "to go to homepage.",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
